src/dmd/cppmanglewin.d

```diff
@@ -684,22 +684,138 @@ private:
             buf.writestring("?1");
             return;
         }
+        else if (sym.ident)
+        {
+            switch (sym.ident.toString)
+            {
+                case "opAssign":    buf.writestring("?4");  return;
+                case "opEquals":    buf.writestring("?8");  return;
+                case "opIndex":     buf.writestring("?A");  return;
+                case "opCall":      buf.writestring("?R");  return;
+                default:            break;
+            }
+        }
         if (TemplateInstance ti = sym.isTemplateInstance())
         {
+            auto id = ti.tempdecl.ident;
+            const(char)* ident = id.toChars;
+
+            // test for special symbols
+            int firstTemplateArg = 0;
+            int whichOp = -1;
+            switch (ti.name.toString)
+            {
+                case "opCast":
+                    buf.writestring("?B");
+                    return;
+                case "opUnary":     whichOp = 0;    break;
+                case "opBinary":    whichOp = 1;    break;
+                case "opOpAssign":  whichOp = 2;    break;
+                case "opAssign":    whichOp = 3;    break;
+                case "opEquals":    whichOp = 4;    break;
+                case "opIndex":     whichOp = 5;    break;
+                case "opCall":      whichOp = 6;    break;
+                default:            break;
+            }
+            if (whichOp != -1)
+            {
+                TemplateDeclaration td = ti.tempdecl.isTemplateDeclaration();
+                assert(td);
+                TemplateParameter tp = (*td.parameters)[0];
+                TemplateValueParameter tv = tp.isTemplateValueParameter();
+                if (tv)
+                {
+                    if (tv.valType.isString())
+                    {
+                        Expression exp = (*ti.tiargs)[0].isExpression();
+                        StringExp str = exp.toStringExp();
+                        switch (whichOp)
+                        {
+                            case 0: // opUnary
+                                switch (str.peekSlice)
+                                {
+                                    case "*":   ident = "?D";   goto continue_template;
```

Comment:
Please do not use `ident` as a pointer to a 0 terminated string. Use `name` instead. Use of `ident` should be restricted to an instance of `Identifier`.